src/DependencyInjection/Compiler/DoctrineOrmMappingsPass.php

```diff
@@ -47,17 +52,27 @@ public function __construct(Definition|Reference $driver, array $namespaces, arr
     }
 
     /**
-     * @param string[]     $namespaces        Hashmap of directory path to namespace.
-     * @param string[]     $managerParameters List of parameters that could which object manager name
-     *                                        your bundle uses. This compiler pass will automatically
-     *                                        append the parameter name for the default entity manager
-     *                                        to this list.
-     * @param string|false $enabledParameter  Service container parameter that must be present to
-     *                                        enable the mapping. Set to false to not do any check,
-     *                                        optional.
+     * @param string[]      $namespaces          Hashmap of directory path to namespace.
+     * @param string[]      $managerParameters   List of parameters that could which object manager name
+     *                                           your bundle uses. This compiler pass will automatically
+     *                                           append the parameter name for the default entity manager
+     *                                           to this list.
+     * @param string|false  $enabledParameter    Service container parameter that must be present to
+     *                                           enable the mapping. Set to false to not do any check,
+     *                                           optional.
+     * @param string[]|bool $enableXsdValidation This should never be an array, but is is due to BC reasons.
      */
-    public static function createXmlMappingDriver(array $namespaces, array $managerParameters = [], string|false $enabledParameter = false, bool $enableXsdValidation = false): self
+    public static function createXmlMappingDriver(array $namespaces, array $managerParameters = [], string|false $enabledParameter = false, bool|array $enableXsdValidation = false): self
     {
+        if (is_array($enableXsdValidation)) {
+            trigger_deprecation('doctrine/doctrine-bundle', '3.2', 'Starting version 3.2 passing $aliasMap to createXmlMappingDriver is deprecated and has no effect.');
```

Comment:
Doctrine has the `doctrine/deprecations` package. The function `trigger_deprecation` is provided by `symfony/deprecation-contracts`.


```suggestion
            \Doctrine\Deprecations\Deprecation::trigger(
                'doctrine/doctrine-bundle',
                'https://github.com/doctrine/DoctrineBundle/pull/2190',
                'Providing a $aliasMap to createXmlMappingDriver is deprecated and has no effect."
            );
```